0xflotus/ip2d
# Add ESLint Configuration and Update TypeScript Settings

### Description
This PR installs ESLint for improved code quality and updates the TypeScript compiler options to enhance type-checking and module compatibility.

### New
- ESLint and TypeScript ESLint plugin have been added for better linting support.

### Improvements
- Enhanced `tsconfig.json` with stricter compiler options for improved type safety and module handling.

### Bugs Resolved
- No specific bugs were addressed; however, the introduction of ESLint will help in identifying and resolving potential code issues moving forward.

### Why This Change is Necessary
The addition of ESLint will promote a consistent coding style and catch potential issues early in the development process. The updated TypeScript settings aim to improve code quality and reduce errors.

### How to Test
1. Run `npm run lint` to check for any linting errors.
2. Verify that the TypeScript compiler options are effective by running `tsc` and ensuring there are no compilation errors.
